os/booting-on-vmware.md

```diff
@@ -58,11 +85,104 @@ retained, although CoreOS supports config-drive on all platforms.
+
+* You can set guestinfo keys and values from the CoreOS guest itself by using preinstalled vmwaretools command like:
+
+```sh
+/usr/share/oem/bin/vmtoolsd --cmd "info-set guestinfo.<variable> <value>"
+```
+
+* You can also set guestinfo keys and values from a Service Console:
+
+```sh
+vmware-cmd /vmfs/volumes/xxxxxxxx.../VMNAME/VMNAME.vmx setguestinfo <variable> <value>
+```
+
+* You can manually modify the VMX and reload it on the VMware Workstation, ESXi host or the vCenter.
+
+* If you change or set GuestInfo settings by using VMware API or CoreOS guest itself, values are only stored in VM process memory. In case of reboot or shutdown, values will be loosed.
+
+You can read more about GuestInfo in [this](http://blog-lrivallain.rhcloud.com/2014/08/15/vmware-use-guestinfo-variables-to-customize-guest-os/) blog post.
+
+### GuestInfo Example
+
+Example below provides hostname, DNS address, interface role, and static `192.168.178.97` IP address with `255.255.255.0` netmask on the Ethernet interface which matches `eno1*` interface name and `00:0c:29:63:92:5c` interface mac address (you can simply provide only one match option: `name` or `mac`):
+
+```
+guestinfo.hostname = "coreos"
+guestinfo.interface.0.role = "private"
+guestinfo.dns.server.0 = "8.8.8.8"
+guestinfo.interface.0.route.0.gateway = "192.168.178.1"
+guestinfo.interface.0.route.0.destination = "0.0.0.0/0"
+guestinfo.interface.0.mac = "00:0c:29:63:92:5c"
+guestinfo.interface.0.name = "eno1*"
+guestinfo.interface.0.dhcp = "no"
+guestinfo.interface.0.ip.0.address = "192.168.178.97/24"
+```
+
+The CoreOS OVA image contains VMware tools and OEM Cloud-Config which executes `coreos-cloudinit` with `--oem=vmware` option. This option automatically sets the `-from-vmware-backdoor` and `-convert-netconf=vmware` flags. Using values above and these flags `coreos-cloudinit` will generate following systemd network unit:
+
+```
+[Match]
+Name=eno1*
+MACAddress=00:0c:29:63:92:5c
+
+[Network]
+DNS=8.8.8.8
+
+[Address]
+Address=192.168.178.97/24
+
+[Route]
+Destination=0.0.0.0/0
+Gateway=192.168.178.1
+```
+
+This unit file will configure the matching network interface. If you have set `coreos.config.data` or `coreos.config.url` variables, `coreos-cloudinit` will try to apply these Cloud Configs on VM and substitute `$private_ipv4` and `$public_ipv4` variables if you've configured network interfaces' roles using `interface.<n>.role` guest variable.
+
+### Defining Cloud-Config in GuestInfo
+
+You have to encode your regular Cloud-Config with `base64` tool and set output into `coreos.config.data` variable.
+
+|  encoding   |                      command                      |
+|:------------|:--------------------------------------------------|
+| base64      | `base64 -w0 /path/to/user_data && echo`           |
+| gzip+base64 | `gzip -c /path/to/user_data | base64 -w0 && echo` |
+
+Or simply set URL in `coreos.config.url` variable.
```

Comment:
To avoid having to base64, you can retrieve your Cloud-Config from a URL with the `coreos.config.url` variable.

question: is this true? or do you need to base64 encode that too?
question: do these variables need `guestinfo.`?